packages/@lwc/integration-karma/scripts/karma-plugins/node-env.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+/**
+ * This transformation replaces:
+ *     process.env.NODE_ENV === 'test-karma-lwc'
+ * with:
+ *     true
+ */
+'use strict';
+
+const MagicString = require('magic-string');
+const Watcher = require('./Watcher');
+
+function createPreprocessor(config, emitter, logger) {
+    const log = logger.create('preprocessor-node-env');
+    const watcher = new Watcher(config, emitter, log);
+
+    return (code, file, done) => {
+        const input = file.path;
+
+        watcher.watchSuite(input, []);
+
+        const magicString = new MagicString(code);
+        magicString.replaceAll(`process.env.NODE_ENV === 'test-karma-lwc'`, `true`);
```

Comment:
After reading a bit more about the PR, I get a better sense for why we have to replace the entire string `process.env.NODE_ENV === 'test-karma-lwc'` instead of just setting `process.env.NODE_ENV` to `test`. 

The gist is that, we want the engine to run as if `process.env.NODE_ENV === 'production'` which is taken care of by karma-plugins/env.js. However, we also want to have some testing related pieces of the engine to run as if `process.env.NODE_ENV === 'test-karma-lwc'`. Hence we cannot just do a `magicString.replaceAll(`process.env.NODE_ENV`, 'test-karma-lwc'`);`

Reply:
Exactly. It's a bit weird, but that was the best solution I could think of.

A more holistic solution would probably be to have Karma run on artifacts that are completely different from what we ship to users. (And thus we can do whatever we want, e.g. `process.env.IS_KARMA`.) But then that would introduce the risk that we aren't testing what we're shipping...

Reply:
Yet another option I considered was to ship code comments with some kind of hint to tell the Karma preprocessor to uncomment it. But that seemed really janky since we wouldn't get linting/typechecking on the code.